document-generator-core/src/test/java/uk/gov/companieshouse/document/generator/core/service/DocumentGeneratorServiceTest.java

```diff
@@ -0,0 +1,180 @@
+import uk.gov.companieshouse.document.generator.core.models.DocumentRequest;
+import uk.gov.companieshouse.document.generator.core.service.impl.DocumentGeneratorServiceImpl;
+import uk.gov.companieshouse.document.generator.core.service.response.ResponseObject;
+import uk.gov.companieshouse.document.generator.core.service.response.ResponseStatus;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoRequest;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DocumentGeneratorServiceTest {
+
+    @Mock
+    private DocumentInfoService mockDocumentInfoService;
+
+    @Mock
+    private RenderDocumentRequestHandler mockRequestHandler;
+
+    @Mock
+    private EnvironmentReader mockEnvironmentReader;
+
+    private DocumentGeneratorService documentGeneratorService;
+
+    private String REQUEST_ID = "requestId";
+
+    private String DESCRIPTION = "description";
+
+    private String DESCRIPTION_IDENTIFIER = "descriptionIdentifier";
+
+    private String SIZE = "size";
+
+    private String LOCATION = "location.file";
+
+    private String DATE = "date";
+
+    private String DATE_VALUE = "01/01/1980";
+
+    @BeforeEach
+    public void setUp() {
+        documentGeneratorService = new DocumentGeneratorServiceImpl(mockDocumentInfoService,
+                mockEnvironmentReader, mockRequestHandler);
+    }
+
+    @Test
+    @DisplayName("Test a successful generate completed")
+    public void testsSuccessfulGenerateCompleted() throws IOException {
+
+        when(mockDocumentInfoService.getDocumentInfo(any(DocumentInfoRequest.class))).thenReturn(setSuccessfulDocumentInfo());
+        when(mockRequestHandler.sendDataToDocumentRenderService(any(String.class), any(RenderDocumentRequest.class))).thenReturn(setSuccessfulRenderResponse());
+
+        ResponseObject response = documentGeneratorService.generate(setValidRequest(), REQUEST_ID);
+
+        assertNotNull(response);
+        assertNotNull(response.getData());
+        assertEquals(response.getData().getDescriptionValues(), setDescriptionValue());
+        assertEquals(response.getData().getDescription(), DESCRIPTION);
+        assertEquals(response.getData().getDescriptionIdentifier(), DESCRIPTION_IDENTIFIER);
+        assertEquals(response.getData().getSize(), SIZE);
+        assertEquals(response.getData().getLinks(), LOCATION);
```

Comment:
These are the wrong way round. The method signature is: assertEquals(expected, actual);